File: src/main/java/me/udnek/fnafu/game/mechanic/door/DoorButton.java
package me.udnek.fnafu.game.mechanic.door;

import me.udnek.fnafu.map.Originable;
import me.udnek.fnafu.map.location.LocationData;
import me.udnek.fnafu.map.location.LocationSingle;
import org.bukkit.Location;

public class DoorButton implements Originable {

    private LocationSingle button;

    public DoorButton(LocationSingle locationSingle){
        button = locationSingle;
    }

    @Override
    public void setOrigin(Location location) {
        button.setOrigin(location);
    }

    public Location getLocation(){
        return button.getFirst();
    }
}

File: src/main/kotlin/me/udnek/fnafu/mechanic/door/DoorButton.kt
package me.udnek.fnafu.mechanic.door

import me.udnek.fnafu.map.location.LocationSingle
import me.udnek.itemscoreu.custom.minigame.Originable
import org.bukkit.Location

class DoorButton(private val button: LocationSingle) : Originable {
    override fun setOrigin(origin: Location) {
        button.setOrigin(origin)
    }

    val location: Location
        get() = button.first.clone()
}
